src/sas/qtgui/Plotting/Slicers/BoxSlicer.py

```diff
@@ -477,18 +577,197 @@ def restore(self, ev):
         """
         Restore the roughness for this layer.
         """
-        self.x = self.save_x
-        self.y = self.save_y
+        self.y2 = self.save_y2
+        self.x2 = self.save_x2
+        self.y1 = self.save_y1
+        self.x1 = self.save_x1
+        self.half_height = self.save_half_height
+        self.half_width = self.save_half_width
 
     def move(self, x, y, ev):
         """
         Process move to a new position, making sure that the move is allowed.
         """
+        self.x1 = x
+        delta = self.x1 - self.center_x
+        self.x2 = self.center_x - delta
+        self.half_width = numpy.fabs(self.x1 - self.x2) / 2
         self.has_move = True
         self.x = x
         self.base.update()
         self.base.draw()
 
+    def setCursor(self, x, y):
+        """
+        Update the figure given x and y
+        """
+        self.move(x, y, None)
+        self.update()
+
+class HorizontalDoubleLine(BaseInteractor):
+    """
+    Draw 2 vertical lines that can move symmetrically in opposite directions in y and centered on
+    a point (PointInteractor). It also defines the left and right x positions of a box.
+    """
+    def __init__(self, base, axes, color='black', zorder=5, x=0.5, y=0.5,
+                 center_x=0.0, center_y=0.0):
```

Comment:
Nitpick: `HorizontalDoubleLine` and `VerticalDoubleLine` signatures are almost exactly the same. A meta class that encompasses this overlap would be useful to minimize code repetition.

Reply:
Agreed @krzywon  - but that is part of a bigger slicer refactor (which I have aspirations for?)  so not for this PR.

BTW - these two interactors classes are also used by boxsum. And when I say "also used" I mean rewritten completely (copy pasted?). 

There are really only a small number of base element interactors (radial line, arc, center, horizontal and vertical lines).  That to say that I have a whole aspirational map drawn out. but that will be a lot of work.